File: api/routers/feedback.py
from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from api.deps import get_db
from api.schemas.feedback import FeedbackListResponse, FeedbackRecord, RecordFeedbackRequest
from api.services import feedback_service

router = APIRouter(prefix="/api/feedback", tags=["feedback"])


@router.post("", response_model=FeedbackRecord, status_code=201)
def record_feedback(
    request: RecordFeedbackRequest, session: Session = Depends(get_db)
) -> FeedbackRecord:
    return feedback_service.record_feedback(session, request)


@router.get("", response_model=FeedbackListResponse)
def list_feedback(
    customer_id: uuid.UUID | None = None,
    limit: int = Query(default=20, le=100),
    offset: int = 0,
    session: Session = Depends(get_db),
) -> FeedbackListResponse:
    return feedback_service.list_feedback(session, customer_id, limit, offset)

File: dashboard/src/api/feedback.ts
import { getJson, postJson } from './client'
import type { FeedbackListResponse, FeedbackRecord, RecordFeedbackRequest } from './types'

export function recordFeedback(request: RecordFeedbackRequest) {
  return postJson<FeedbackRecord>('/api/feedback', request)
}

export interface ListFeedbackParams {
  customer_id?: string
  limit?: number
}

export function listFeedback(params: ListFeedbackParams = {}, signal?: AbortSignal) {
  const search = new URLSearchParams()
  if (params.customer_id) search.set('customer_id', params.customer_id)
  search.set('limit', String(params.limit ?? 20))
  return getJson<FeedbackListResponse>(`/api/feedback?${search.toString()}`, signal)
}
